### ISSUE TITLE
Build fails, branch is dirty w/ semantic versioning - solved

### ISSUE THREAD

**Issue by zment4** [game-ci/unity-builder#537]
I was using GitHub Actions with the gameci unity builders, and I had an issue with the build not going forward as it complained that the branch is dirty. After googling around for a while I came across this #241 where the op had nearly the same problem as me. The workaround was to use allowDirtyBuild, which wasn't recommended, so I tried to look for something else going wrong. I spotted a mention that the error message should tell me what files are the dirty ones, and so I went and looked at the error message a bit closer. 

Apparently the semantic versioning build script creates an Editor folder under Assets (presumably for a custom build script). Now, I don't have an Editor folder inside Assets, as I haven't had the need for one (yet). Unity then creates an Editor.meta for that folder, and whatever method you use to sense modifications that won't dirty the repository (if it's simply .gitignore, then yeah you can't really add Editor.meta to .gitignore), doesn't realize that the build script was the reason the repo was dirtied.

So the workaround was to create Editor folder in the Asset folder for my project, and commit the Editor.meta. And thus the build went forward.

I'm posting this here in case someone else is having the same problem, as there wasn't a lot of relevant info about this (at least that I found) except issue #241 as previously mentioned. Also, sorry if I categorized this wrong, as I'm not really sure if this is a bug, but it's something that isn't (at least obviously - or I just missed it) said in the documentation either.

**Comment by webbertakken** [comment]
Hey @zment4,

Thank you for your input. This is the right place :)

The Editor folder with the custom build script is placed there (in the docker step) after the dirty check (part of the javascript preparation) if I remember correctly.

Are you running multiple builds in succession on the same machine without clearing any files?

Have you read our [docs](https://game.ci/docs/troubleshooting/common-issues#branch-is-dirty) on this common issue?

**Comment by zment4** [comment]
I can't be totally sure I've scoured the docs completely, but I did spend a few hours trying to figure this out, of which a big chunk of time was trying to read the docs. Thanks for the link, tho I had read that before, and at least I thought I understood what it was trying to say...

...HOWEVER. You might be on to something with the multiple builds thingy. I'm using GitHub Actions, and as I followed the suggestions on the docs (and on the other issue I linked) to check what files are deemed dirty and try to fix the repo with that, I _simultaneously_ modified the workflow. If I remember correctly, I think I had two build tasks running simultaneously, and there was some other error on the other build task because of it, so I added concurrency to the jobs so that the builds wouldn't be trying to go about it at the same time.

So that might've been the real fix - if you have multiple build jobs in the same workflow, make sure they use the same concurrency such that they aren't activated at the same time!

Thanks for the response!
